src/main/java/io/github/ust/mico/kafkafaasconnector/MessageListener.java

```diff
@@ -260,4 +267,23 @@ private ErrorReportMessage getErrorReportMessage(String message, String original
     public boolean isTestMessageCompleted(MicoCloudEventImpl<JsonNode> cloudEvent, String topic) {
         return cloudEvent.isTestMessage().orElse(false) && topic.equals(cloudEvent.getFilterOutBeforeTopic().orElse(null));
     }
+
+    /**
+     * Sets the time, the correlationId and the Id field of a CloudEvent message if missing
+     * @param cloudEvent
+     * @param originalMessageId
+     */
+    public void setMissingHeaderFields(MicoCloudEventImpl<JsonNode> cloudEvent, String originalMessageId){
+        if(StringUtils.isEmpty(cloudEvent.getId())){
+            cloudEvent.setRandomId();
+            log.debug("Added missing id '{}' to cloud event", cloudEvent.getId());
+        }
+        if(!cloudEvent.getTime().isPresent()){
```

Comment:
use isEmpty instead of negation for better visibility of negation

Reply:
there is no isEmpty for a java [optional](https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html) and `ifPresentOrElse` which would help here is Java 9+.

Reply:
My ide says otherwise and it compiles and runs all tests flawless...
I don't know why since your source should be right.

Reply:
isEmpty is added to Java in version 11. We could use it, if we decide to update to java 11+.